S/SuiteSparse/SuiteSparse@7/build_tarballs.jl

```diff
@@ -37,29 +37,32 @@ else
     )
 fi
 
-for proj in SuiteSparse_config AMD BTF CAMD CCOLAMD COLAMD CHOLMOD LDL KLU UMFPACK RBio SPQR; do
-    cd ${proj}/build
-    cmake .. -DCMAKE_BUILD_TYPE=Release \
-             -DCMAKE_INSTALL_PREFIX=${prefix} \
-             -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
-             -DENABLE_CUDA=0 \
-             -DNFORTRAN=1 \
-             -DNOPENMP=1 \
-             -DNPARTITION=0 \
-             -DNSTATIC=1 \
-             -DBLAS_FOUND=1 \
-             -DBLAS_LIBRARIES="${libdir}/lib${BLAS_NAME}.${dlext}" \
-             -DBLAS_LINKER_FLAGS="${BLAS_NAME}" \
-             -DBLAS_UNDERSCORE=ON \
-             -DBLA_VENDOR="${BLAS_NAME}" \
-             -DLAPACK_FOUND=1 \
-             -DLAPACK_LIBRARIES="${libdir}/lib${BLAS_NAME}.${dlext}" \
-             -DLAPACK_LINKER_FLAGS="${BLAS_NAME}" \
-             "${CMAKE_OPTIONS[@]}"
-    make -j${nproc}
-    make install
-    cd ../..
-done
+PROJECTS_TO_BUILD="suitesparse_config;amd;btf;camd;ccolamd;colamd;cholmod;klu;ldl;umfpack;rbio;spqr"
+
+cmake -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=${prefix} \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DBUILD_STATIC_LIBS=OFF \
+      -DBUILD_TESTING=OFF \
+      -DSUITESPARSE_ENABLE_PROJECTS=${PROJECTS_TO_BUILD} \
+      -DSUITESPARSE_DEMOS=OFF \
+      -DSUITESPARSE_USE_STRICT=ON \
+      -DSUITESPARSE_USE_CUDA=OFF \
+      -DSUITESPARSE_USE_FORTRAN=OFF \
+      -DSUITESPARSE_USE_OPENMP=OFF \
+      -DNPARTITION=OFF \
+      -DBLAS_FOUND=1 \
+      -DBLAS_LIBRARIES="${libdir}/lib${BLAS_NAME}.${dlext}" \
+      -DBLAS_LINKER_FLAGS="${BLAS_NAME}" \
+      -DBLAS_UNDERSCORE=ON \
+      -DBLA_VENDOR="${BLAS_NAME}" \
```

Comment:
`BLA_VENDOR` is a typo?

Reply:
No, SuiteSparse references this to determine the vendor of the BLAS, so since we supply all the flags ourselves, we need to set it ourselves.

This lets it print it out in the CMake log as well, e.g. https://buildkite.com/julialang/yggdrasil/builds/7473#018cc579-e8d8-4282-b92c-fef9a361a31b/630-935 has
```
[14:45:18] -- Found blastrampoline 64-bit BLAS
[14:45:18] -- BLAS64_suffix: _64
[14:45:18] -- BLAS64 suffix has an underscore
[14:45:18] -- BLAS libraries:      /workspace/destdir/lib/libblastrampoline.dylib
[14:45:18] -- BLAS linker flags:   blastrampoline
[14:45:18] -- BLAS include:
```

Reply:
I was just wondering whether the name `BLA` instead of `BLAS` was a typo on our side, but it's the name used upstream.

Reply:
Yea, everything starting with `BLAS` is the CMake-internal variables, so SuiteSparse used the `BLA` suffix. I think I recall a comment recently about how that is confusing for other people, but it doesn't look like this was renamed during the recent cleanup of the options.